### PR TITLE
Auto generate documentation for preferences

### LINKED ISSUE #2564: Add documentation for preferences
## 📚 Documentation

Add initial documentation for preferences

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

Fixes https://github.com/napari/napari/issues/2564

The `preferences.md` is autogenerated from the CORE_SETTINGS.

## Type of change

- [x] This change requires a documentation update

# References

# How has this been tested?

- [x] example: all tests pass with my change

## Image

Visit https://github.com/napari/napari/blob/37035457cedbf4431dd32b4b36e04ce490d2788b/docs/guides/preferences.md

image

### CODE REVIEW

**Comment by sofroniewn** [comment]
@goanpeca i noticed this PR still has some png inside it. I was hoping that with the "auto generation" we wouldn't need to save pngs but instead be able to screenshot the preferences dialog and add it as @tlambert03 has done in other places. Maybe I'm not quite understanding or he can advise. I'll take a look at the text now too

**Comment by goanpeca** [comment]
The pngs are autogenerated also, from the same script as well as the md file.

**Comment by sofroniewn** [comment]
> The pngs are autogenerated also, from the same script as well as the md file.

Ah interesting, thanks, do you think we can avoid saving the pngs and committing them to the repo though?

I know it the past @jni has been worried about adding image files to the repo as the repo history size starts increasing a lot. That might have applied more to our tutorials and examples where files could get very large very quickly. It looks like these are only 200kB, so will take a while for them to accumulate to get a very large history (note my mental model right now is if we do this 100 times, even if at one time the repo only has 200kB the history will have 20MB).

If the history size problem isn't a concern then maybe this is fine. Definitely good that they are autogenerated by the script. We could also look into another repo/ place on github to host the files from the script too (which is a little more complicated, but I know github want to make that easier)

**Comment by goanpeca** [comment]
Agreed @sofroniewn 

Open to suggestions 🙂

**Comment by sofroniewn** [comment]
> Open to suggestions 🙂

@jni @tlambert03 do you have any suggestions on this one?

**Review by tlambert03 — COMMENTED** [review]
love this @goanpeca.  want to see more of it! :)

regarding the png files, since you're already autogenerating them, i think the best thing to do here is probably to have them be created in a subfolder `docs/images/_autgenerated` which is added to `.gitignore` ... and then either just add `python tools/update_docs.py` to our `make_docs.yml` in github in the build docs step:

https://github.com/napari/napari/blob/7b92de035a0a38a0c8f325e2336c39407b4d0822/.github/workflows/make_docs.yml#L78-L80

... or, update the makefile `make docs` command and switch out the github action to use that as well.

two other minor comments:

1. for the type field, you might use `{class}`{{ fields["type"].__module__ }}.{{ fields["type"].__name__ }}` ... instead of `{{ fields["type"] }}`.  (will render a bit nicer, and will also link to the API docs if they are ever exposed)

2. I find the text (fonts) a bit hard to visually parse:

Untitled

 I wonder if perhaps italicizing `{{ section_data["description"] }}` might help... and then either reducing the font-size, or color, or indenting the bullet points slightly?  I know that's not really the main point of this PR, but just a thought

**Comment by goanpeca** [comment]
Made the suggested changes @tlambert03, thanks!

File: docs/guides/preferences.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
+(preferences)=
```
Comment (tlambert03) [inline docs/guides/preferences.md]: should this be checked into source?

Comment (goanpeca) [reply docs/guides/preferences.md]: Good question I am fine with any option. @sofroniewn @jni ?

Comment (sofroniewn) [reply docs/guides/preferences.md]: Is this the file that gets autogenerated? Is this the version "before autogeneration" that get's overwritten? I don't have a solid opinion, could go either way - curious what people think are pros/ cons?

Comment (jni) [reply docs/guides/preferences.md]: If this is generated I prefer not to commit it.

Comment (goanpeca) [reply docs/guides/preferences.md]: >Is this the file that gets autogenerated? 

Yes @sofroniewn

Comment (sofroniewn) [reply docs/guides/preferences.md]: Then I am fine with @jni preference not to commit it. Don't have strong views though

Comment (sofroniewn) [reply docs/guides/preferences.md]: If you can drop this file @goanpeca i think this is then good to merge

Comment (goanpeca) [reply docs/guides/preferences.md]: Fixed!

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**